### PR TITLE
[Cache] Dont use Redis connection when not required

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 3.1 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/RedisAdapter.php [hunk 1]
```diff
@@ -19,30 +19,34 @@
...
 
     public function __construct(\Redis $redisConnection, $namespace = '', $defaultLifetime = 0)
     {
-        $this->redis = $redisConnection;
-
         if (preg_match('#[^-+_.A-Za-z0-9]#', $namespace, $match)) {
             throw new InvalidArgumentException(sprintf('RedisAdapter namespace contains "%s" but only characters in [-+_.A-Za-z0-9] are allowed.', $match[0]));
         }
+        $this->redis = $redisConnection;
+        $this->redis->setOption(\Redis::OPT_PREFIX, $namespace);
```
Comment (stof) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: this breaks things if the same Redis connection is reused for several things, as you change its state

File: src/Symfony/Component/Cache/Adapter/RedisAdapter.php [hunk 2]
```diff
@@ -62,14 +66,14 @@ protected function doHave($id)
      */
     protected function doClear($namespace)
     {
-        if (!isset($namespace[0])) {
+        if (!isset($this->namespace[0])) {
```
Comment (stof) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: why changing this ?

File: src/Symfony/Component/Cache/Adapter/RedisAdapter.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: true, reverted

File: src/Symfony/Component/Cache/Adapter/RedisAdapter.php [hunk 3]
```diff
@@ -36,13 +35,15 @@ public function __construct(\Redis $redisConnection, $namespace = '', $defaultLi
...
     protected function doFetch(array $ids)
     {
-        $values = $this->redis->mget($ids);
-        $index = 0;
-        $result = [];
-
-        foreach ($ids as $id) {
-            if (false !== $value = $values[$index++]) {
-                $result[$id] = unserialize($value);
+        $result = array();
```
Comment (Nek-) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: As it is for Sf 3.1 brackets were great, isn't it ?

Comment (stof) [inline src/Symfony/Component/Cache/Adapter/RedisAdapter.php]: @Nek- our policy is to not use them, to keep consistency in the codebase (and we cannot convert the whole codebase to the short array syntax, as we will need to merge 2.8 into 3.x during 3 years, and we don't want git conflicts each time we change a line dealing with an array)

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**